### PR TITLE
Enable custom CSR signer name

### PR DESCRIPTION
**Pull request by rdvencioneck**
Some Control planes, such as AWS EKS, won't accept `kubernetes.io/kubelet-serving` as the CSR SignerName.
As documented [here](https://docs.aws.amazon.com/eks/latest/userguide/cert-signing.html), EKS users can use `beta.eks.amazonaws.com/app-serving` as the SignerName, instead, to get the certificate properly signed by the control plane.

This PR will enable any custom SignerName, so that it can work for EKS or other managed control planes that may require custom SignerName. Solves https://github.com/admiraltyio/admiralty/issues/120

### CODE REVIEW

**Comment by adrienjt** [comment]
Thank you @rdvencioneck and sorry for the late reply. Could you please add the field to the [chart's readme](https://github.com/admiraltyio/admiralty/blob/master/charts/multicluster-scheduler/README.md).

And ideally a note or page in the [operator guide](https://github.com/admiraltyio/admiralty/tree/master/docs/operator_guide). I'm sure a lot of people will find this useful.

Out of scope for this PR: a subsequent PR would be welcome to run the end-to-end tests on EKS, to prove that this works and continues to work.

**Comment by rdvencioneck** [comment]
hi @adrienjt 
sorry, I missed your review.
I've included the requested info and will take a look at the referenced PR after this one is merged

Thanks!

File: cmd/agent/main.go [hunk 1]
```diff
@@ -358,6 +360,7 @@ func parseFlags() *options {
 }
 
 func setupLogging(ctx context.Context, o *options) {
+	log.SetLogger(zap.New(zap.UseDevMode(true)))
```
Comment (adrienjt) [inline cmd/agent/main.go:363]: I know logging isn't great, but could you keep it as is for now?

FYI, I have a draft to improve it: https://github.com/admiraltyio/admiralty/pull/214

**Review by adrienjt — CHANGES_REQUESTED** [review]

File: charts/multicluster-scheduler/README.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ Admiralty uses [finalizers](https://kubernetes.io/docs/tasks/access-kubernetes-a
 | controllerManager.securityContext | object | `{}` |  |
 | controllerManager.affinity | object | `{}` |  |
 | controllerManager.tolerations | array | `[]` |  |
+| certificateSignerName | string | `"kubernetes.io/kubelet-serving"` | SignerName for the virtual-kubelet certificate signing request
```
[the remark this answers is not in the document]
Comment (rdvencioneck) [reply charts/multicluster-scheduler/README.md]: fixed!

File: docs/operator_guide/authentication.md [hunk 3]
```diff
@@ -111,3 +111,10 @@ spec:
...
 
 The `prefix` field is important. It is combined with client certificate common names (over which this cluster has no control) to form Kubernetes user names, which are matched to RBAC rules. To avoid unwanted impersonation, it is recommended to use prefixes to isolate TrustedIdentityProviders by trust domain, typically one prefix by TrustedIdentityProvider. By convention (following the [SPIFFE](https://spiffe.io/) standard used by other mTLS projects like service meshes), trust domain prefixes usually start with `spiffe://` and end with a forward slash. Combined with Identity certificate common names (see above), Kubernetes user names look like `spiffe://source-cluster/ns/namespace-a/id/default`. Use them as subjects of RBAC rules to authorize requests. For multi-cluster scheduling, Admiralty generates the right RBAC rules from [Source and ClusterSource](scheduling.md#sources-and-cluster-sources) objects.
+
+
+## Virtual Kubelet certificate
```
[the remark this answers is not in the document]
Comment (rdvencioneck) [reply docs/operator_guide/authentication.md]: makes sense. moved

File: cmd/agent/main.go (hunk 1 above)
Comment (rdvencioneck) [reply cmd/agent/main.go]: removed

**Review by adrienjt — APPROVED** [review]

**Comment by rdvencioneck** [comment]
@adrienjt could you please trigger the tests again? 
They passed previously, but auto-merge didn't work for some reason

**Pull request merged**